package/MDAnalysis/core/selection.py

```diff
@@ -655,8 +655,14 @@ def _apply(self, group):
         pattern = Chem.MolFromSmarts(self.pattern)
         if not pattern:
             raise ValueError(f"{self.pattern!r} is not a valid SMARTS query")
+        use_chirality = self.rdkit_kwargs.pop("useChirality", True)
+        max_matches = self.rdkit_kwargs.pop("maxMatches", group.n_atoms)
```

Comment:
my comment was more to have a separate kwarg from `rdkit_kwargs` so that folks don't get confused because what's being passed to what - thoughts @lilyminium ?

The useChirality being inverse of the default here is kind of a pain :/

Reply:
Yeah, I guess something like `rdkit_kwargs` and `substruct_match_kwargs` could work better -- I defer to @cbouy's opinions on casing and spelling of the latter. it is nice to be consistent but usually means I just google "rdkit substruct match" instead of "mdanalysis substruct match".

> The useChirality being inverse of the default here is kind of a pain :/

@cbouy could you comment on this? I also usually set it to True when matching, I haven't really thought through the argument for just defaulting to RDKit's defaults when possible, vs. returning what we think are the best results by default.

Reply:
I think I'd rather have a `smarts_kwargs` that gets passed to RDKit's `GetSubstructMatches` and a `rdkit_kwargs` (or another more descriptive name) that get's passed to `convert_to` instead of having a kwarg that combines both. Sounds more future-proof to me.

@lilyminium Users might not know that when they type `u.select_atom("smarts ...")` the underlying function being called is `GetSubstructMatch`, which is why I would go with `smarts_kwargs` instead of `substruct_match_kwargs`.

Googling "mdanalysis smarts" will take users to the docs page so @orioncohen should add a link to RDKit's [SubstructMatch docs](https://www.rdkit.org/docs/source/rdkit.Chem.rdchem.html#rdkit.Chem.rdchem.Mol.GetSubstructMatches) somewhere in the `select_atoms` docstring.

Also, we had started a similar discussion in #3324 and we ended up using `rdkit_kwargs` but I don't consider it to be set in stone. I'm not sure if converters other than RDKit are being used for making selections as of now, but I guess it will likely happen in the future so renaming it to `converter_kwargs` should be out of the question.
As @richardjgowers pointed out in the other PR, people using a SMARTS query might not know we implemented it with RDKit and that we need to convert the atomgroup to an RDKit molecule (even if it's documented) so `rdkit_kwargs` might not be descriptive enough. `rdkit_converter_kwargs` is a bit long, so `rdkit_converter` maybe ?

As for `useChirality`, I probably went for the non-default because we didn't expose the args for substructure matching at that time so I went for the option that would work for everyone. Also, not sure why it's not the default in RDKit actually, maybe a performance reason ? Anyway we could remove it now, and indicate in the docs that chirality is ignored by default when performing the substructure search.

Reply:
Good point. It is definitely more future proof. I went with `smarts_kwargs` and `rdkit_kwargs` for now. I also added a link to the RDkit documentation as you suggested.

Currently I have `useChirality` set to `True`, but I'll change it based on the outcome of this discussion. If it is set to `True` and a user supplies a stereochemistry agnostic query, such as "CCC(C)O" instead of "CC[C@H](C)O" or "CC[C@@H](C)O", will that select both stereoisomers?

Reply:
Yes it selects both